src/identity/mod.rs

```diff
@@ -0,0 +1,111 @@
+mod platform;
+
+use crate::config::inputs;
+use failure::{Fallible, ResultExt};
+use serde::Serialize;
+use std::collections::HashMap;
+
+/// Kernel arguments location
+static KERNEL_ARGS_FILE: &str = "/proc/cmdline";
+
+/// Agent identity.
+#[derive(Debug, Serialize)]
+pub(crate) struct Identity {
+    /// Collecting level
+    pub(crate) level: String,
+    /// OS platform.
+    pub(crate) platform: String,
+}
+
+impl Identity {
+    /// Create from configuration.
+    pub(crate) fn new(cfg: &inputs::CollectingInput) -> Fallible<Self> {
+        let collecting_level = &cfg.level;
+        let id = match collecting_level.as_str() {
+            "minimal" => Self::try_default("minimal").context("failed to build minimal (default) identity")?,
+            "full" => Self::try_default("full").context("failed to build full identity")?,
+            &_ => Self::try_default("minimal").context("failed to build minimal (default) identity")?,
+        };
+
+        Ok(id)
+    }
+
+    /// Try to fetch default data
+    pub fn try_default(level: &str) -> Fallible<Self> {
+        let platform = platform::read_id(KERNEL_ARGS_FILE)?;
+
+        let id = match level {
```

Comment:
I think you can factor out a bunch of stuff from here, like this:
```rust
let level = match input {
  "minimal" | "full" => input.to_string(),
 _ => "minimal".to_string(),
};
```

Reply:
Yes, that should be cleaner :+1: